Sources/HipMobileUI/ViewModels/Views/AchievementsScreenViewModel.cs

```diff
@@ -44,12 +45,23 @@ public AchievementsScreenViewModel()
                     AchievementNotification.QueueAchievementNotifications(newlyUnlocked);
                 });
 
-                foreach (var achievement in DbManager.DataAccess.Achievements().GetAchievements())
+                //Every second time no achievements are received from the database
+                //If no achievements are received, this method is called again
+                var currentAchievements = DbManager.DataAccess.Achievements().GetAchievements();
+                if (currentAchievements.Any())
                 {
-                    Achievements.Add(AchievementViewModel.CreateFrom(achievement));
+                    foreach (var achievement in currentAchievements)
+                    {
+                        Achievements.Add(AchievementViewModel.CreateFrom(achievement));
+                    }
+
+                    Score = $"{Strings.AchievementsScreenView_Score} {AppSharedData.CurrentAchievementsScore()}";
+                }
+                else
+                {
+                    await UpdateAchievements();
```

Comment:
This seems more like a workaround rather than a solution to the underlying problem. Let's see on Monday if we can find the root cause and eliminate the need for such a workaround.

Reply:
Yeah, this was the only working "solution" I could find for now. We can take look at it tomorrow.